storage/blackhole/ha_blackhole.cc

```diff
@@ -120,7 +121,8 @@ int ha_blackhole::update_row(const uchar *old_data, uchar *new_data)
 {
   DBUG_ENTER("ha_blackhole::update_row");
   THD *thd= ha_thd();
-  if (is_slave_applier(thd) && thd->query() == NULL)
+  if (is_slave_applier(thd) &&
```

Comment:
Please extract this condition to a helper method

Reply:
fixed